opendc-workflow/opendc-workflow-service/src/main/kotlin/org/opendc/workflow/service/internal/StateHelpers.kt

```diff
@@ -0,0 +1,81 @@
+package org.opendc.workflow.service.internal
+
+import org.opendc.workflow.api.WORKFLOW_TASK_DEADLINE
+import org.opendc.workflow.api.WORKFLOW_TASK_MINIMAL_START_TIME
+import org.opendc.workflow.api.WORKFLOW_TASK_SLACK
+import java.util.UUID
+import kotlin.collections.HashMap
+import kotlin.collections.HashSet
+import kotlin.math.max
+
+public fun JobState.computeSlack() {
+    this.computeMinimalStartTimes()
+    for (t in this.tasks) {
+        if (t.dependencies.isEmpty())  {
+            t.task.metadata[WORKFLOW_TASK_SLACK] = 0L
+        } else {
+            val cmin = t.dependencies.minWith(Comparator.comparingLong {it.task.metadata[WORKFLOW_TASK_SLACK] as Long})
```

Comment:
The IDE doesn't give an error, but it.task.metadata does not exist and will result in a NULL pointer exception at runtime.